### PR TITLE
Update material theme

### LINKED ISSUE #448: search for "sendrawtransaction" doesn't return all results
## Describe the bug

The 2 results returned are both for eea_sendRawTransaction. The pages with eth_sendRawTransaction (eg https://besu.hyperledger.org/en/stable/Reference/API-Methods/#eth_sendrawtransaction) do not appear in search results.

Check the issue type in the following list (insert X instead of space between [ ]): -->

- [ ] Missing content
- [ ] Outdated content
- [ ] Wrong content
- [ ] Confusing or misleading content
- [x] Other

## The broken page
Could be an issue with indexing this page:
https://besu.hyperledger.org/en/stable/Reference/API-Methods/#eth_sendrawtransaction

### PR DESCRIPTION
**Pull request by NicolasMassart**
## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [x] you fixed all the issues raised by the tests, if any.
- [x] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change

- upgrade material theme to 6.1.7
- update overrides of templates
- fix search #448 at the same time by updating separators and adding a query replacement script to ignore special chars like `_` `-` or `.`

## Issue fixed

fixes #226
fixes #448 

### For content changes

- [ ] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [x] MkDocs templates
- [x] MkDocs configuration
- [x] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing

See https://hyperledger-besu--573.org.readthedocs.build/en/573/

### Testing the templates:

Look at the header, footer, navs, everything strange you could notice (it should be identical to the current latest version except added the copyright of the template in the footer.

### Testing the search

try words with `-` and `_` like `eth_sendrawtransaction` or `graphql-http-cors-origins

## Screenshots / recording

Search results should look like this:

image

image

### CODE REVIEW

File: custom_theme/main.html [hunk 1]
```diff
@@ -1,106 +1,50 @@
...
-  <!--The button has to point to the doc site and as it's not designed to have doc outside of code
-  repos we have to replace the mkdocs generated link based on the code repos with the doc repos-->
-  {% if page.edit_url and config.extra.doc_site_edit_url %}
-  <a href="{{ page.edit_url | replace(config.repo_url, config.extra.doc_site_edit_url) }}"
-     title="{{ lang.t('edit.link.title') }}"
-     class="md-icon md-content__icon">&#xE3C9;<!-- edit --></a>
+{% block config %}
+  {% if config.plugins.search %}
+    <script>
+      // Make the serach query replace _-. by spaces to extend the search to all RPC methods
```
Comment (macfarla) [inline custom_theme/main.html:25]: s/serach/search

File: custom_theme/partials/footer.html [hunk 2]
```diff
@@ -1,25 +1,3 @@
...
-
-  Permission is hereby granted, free of charge, to any person obtaining a copy
-  of this software and associated documentation files (the "Software"), to
-  deal in the Software without restriction, including without limitation the
-  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
-  sell copies of the Software, and to permit persons to whom the Software is
-  furnished to do so, subject to the following conditions:
-
-  The above copyright notice and this permission notice shall be included in
-  all copies or substantial portions of the Software.
```
Comment (macfarla) [inline custom_theme/partials/footer.html:12]: do we not still need this copyright notice?

**Review by macfarla — COMMENTED** [review]
the search improvement is excellent!

File: custom_theme/partials/footer.html (hunk 2 above)
Comment (NicolasMassart) [reply custom_theme/partials/footer.html:12]: restored

**Review by macfarla — COMMENTED** [review]
LGTM

**Review by MadelineMurray — APPROVED** [review]

**Pull request merged**